src/multizimbutton.cpp

```diff
@@ -165,9 +165,21 @@ ZimItemWidget::ZimItemWidget(QString text, QIcon icon, QWidget *parent) :
     const int border = CSS::MultiZimButton::QListWidget::item::border;
     const int scrollerWidth = CSS::MultiZimButton::QScrollBar::width;
     const int contentWidthExcludeText = iconAndCheckerWidth + totalSpacing + scrollerWidth + border;
-
-    textLabel->setFixedWidth(menu->width() - contentWidthExcludeText);
-    textLabel->setText(text);
+    const int labelWidth = menu->width() - contentWidthExcludeText;
+    textLabel->setFixedWidth(labelWidth);
+
+    QFontMetrics metrics(textLabel->font());
+    const int elideMarkerLength = metrics.tightBoundingRect("(...)").width();
+    const int textLength = labelWidth - elideMarkerLength;
+    QString elidedText = metrics.elidedText(text, Qt::ElideRight, textLength);
+    if (elidedText != text)
+    {
+        /* Remove built-in elide marker */
+        elidedText.chop(1);
+        textLabel->setText(elidedText.trimmed() + "(...)");
+    }
+    else
+        textLabel->setText(text);
```

Comment:
Is there code duplication with custom elision of suggestions?

Reply:
Before the if statement yes. The handling of elide is different since Qt handles LineEdit differently among other text displaying widget (quiet a pain)

Reply:
Done